src/utils/compile.rs

```diff
@@ -0,0 +1,180 @@
+//! Utils to compile script to bytecode and run script from bytecode
+
+use libquickjs_sys as q;
+use crate::bindings::{OwnedValueRef, ContextWrapper};
+use crate::ExecutionError;
+use std::ffi::CString;
+
+/// compile a script, will result in a JSValueRef with tag JS_TAG_FUNCTION_BYTECODE or JS_TAG_MODULE.
+///  It can be executed with run_compiled_function().
+#[allow(dead_code)]
+pub fn compile<'a>(context: &'a ContextWrapper, script: &str, file_name: &str) -> Result<OwnedValueRef<'a>, ExecutionError> {
+    let filename_c = CString::new(file_name).map_err(|_e| { ExecutionError::Internal("cstring creation failed".to_string()) })?;
+    let code_c = CString::new(script).map_err(|_e| { ExecutionError::Internal("cstring creation failed".to_string()) })?;
+
+    let value_raw = unsafe {
+        q::JS_Eval(
+            context.context,
+            code_c.as_ptr(),
+            script.len() as _,
+            filename_c.as_ptr(),
+            q::JS_EVAL_FLAG_COMPILE_ONLY as i32,
+        )
+    };
+
+    // check for error
+    let ret = OwnedValueRef::new(
+        context,
+        value_raw,
+    );
+    if ret.is_exception() {
+        let ex_opt = context.get_exception();
+        if let Some(ex) = ex_opt {
+            Err(ex)
+        } else {
+            Err(ExecutionError::Internal("Unknown error".to_string()))
```

Comment:
This logic is already done by `get_exception()`, this checking is redundant. 
If `get_exception()` is `None`, so you can just do `context.get_exception()?` to get the same outcome.